### PR TITLE
0.4 bindings deprecation

### PR DESCRIPTION
**Pull request by yuyichao**

### CODE REVIEW

**Comment by yuyichao** [comment]
May require https://github.com/JuliaLang/Compat.jl/pull/139 on 0.3.

**Comment by yuyichao** [comment]
CI restarted. Should still wait for a new tag in Compat.

**Comment by yuyichao** [comment]
Bump Compat requirement to 0.7.1

File: README.md [hunk 1]
```diff
@@ -33,9 +33,9 @@ JSON.json(j)
 ## Documentation
 
 ```julia
-JSON.print(io::IO, s::String)
-JSON.print(io::IO, s::Union(Integer, FloatingPoint))
-JSON.print(io::IO, n::Nothing)
+JSON.print(io::IO, s::AbstractString)
+JSON.print(io::IO, s::Union(Integer, AbstractFloat))
+JSON.print(io::IO, n::Void)
```
Comment (stevengj) [inline README.md]: missing `@compat`

**Comment by stevengj** [comment]
Looks good to merge once Travis is green.

**Pull request merged**